youtube2zim/scraper.py

```diff
@@ -608,47 +607,97 @@ def download_from_cache(self, key, video_path):
+            )
+        src_path = files[0]
+
+        # resize thumbnail. we use max width:248x187px in listing
+        # but our posters are 480x270px
+        resize_image(
+            src_path.parent.joinpath("video.jpg"), width=480, height=270, method="cover"
+        )
+
+        # don't reencode if not requesting low-quality and received wanted format
+        if skip_recompress or (
+            not self.low_quality and src_path.suffix[1:] == self.video_format
+        ):
+            return
+
+        dst_path = src_path.parent.joinpath(f"video.{self.video_format}")
+        reencode(
+            src_path, dst_path, enc.to_ffmpeg_args(), delete_src=True, failsafe=False
+        )
+
     def download_video_files_batch(self, options, videos_ids):
         succeeded = []
         failed = []
+
+        # set encoders
+        if self.video_format == "mp4":
+            enc = VideoMp4Low()
+        else:
+            enc = VideoWebmLow()
+
         for video_id in videos_ids:
             options_copy = options.copy()
             video_location = options_copy["y2z_videos_dir"].joinpath(video_id)
             downloaded_from_cache = False
             if self.s3_storage:
                 s3_key = f"{self.video_format}/{self.video_quality}/{video_id}"
                 video_path = video_location.joinpath(f"video.{self.video_format}")
-                downloaded_from_cache = self.download_from_cache(s3_key, video_path)
+                downloaded_from_cache = self.download_from_cache(
+                    s3_key, video_path, enc.VERSION
+                )
                 # option to skip download of video, but not thumbnail and subtitles
                 if downloaded_from_cache:
                     options_copy["skip_download"] = True
             try:
                 with youtube_dl.YoutubeDL(options_copy) as ydl:
                     ydl.download([video_id])
-                post_process_video(
+                self.post_process_video(
                     video_location,
                     video_id,
-                    options_copy["y2z_video_format"],
-                    options_copy["y2z_low_quality"],
+                    enc,
                     skip_recompress=downloaded_from_cache,
                 )
                 succeeded.append(video_id)
-            except (youtube_dl.utils.DownloadError, FileNotFoundError):
+            except (
+                youtube_dl.utils.DownloadError,
+                FileNotFoundError,
+                subprocess.CalledProcessError,
+            ):
                 failed.append(video_id)
-            if self.s3_storage and not downloaded_from_cache:
-                self.upload_to_cache(s3_key, video_path)
+            else:  # upload to cache only if everything went well
+                if self.s3_storage and not downloaded_from_cache:
+                    self.upload_to_cache(s3_key, video_path, enc.VERSION)
```

Comment:
we used to send `"v1"` so sending an int here breaks the S3 storage.
Also as this is a metadata, it must be a string. You may change that in `upload_to_cache`